integration/test/Test/Conversation.hs

```diff
@@ -0,0 +1,145 @@
+module Test.Conversation where
+
+import qualified API.BrigInternal as Internal
+import qualified API.GalleyInternal as API
+import Control.Applicative
+import qualified Data.Aeson as Aeson
+import GHC.Stack
+import SetupHelpers
+import Testlib.Prelude
+
+testDynamicBackendsFullyConnectedWhenAllowAll :: HasCallStack => App ()
+testDynamicBackendsFullyConnectedWhenAllowAll = do
+  let overrides =
+        def {dbBrig = setField "optSettings.setFederationStrategy" "allowAll"}
+  startDynamicBackends [overrides, overrides, overrides] $ \dynDomains -> do
+    [domainA, domainB, domainC] <- pure dynDomains
+    uidA <- randomUser domainA def {Internal.team = True}
+    uidB <- randomUser domainA def {Internal.team = True}
+    uidC <- randomUser domainA def {Internal.team = True}
+    let assertConnected u d d' =
+          bindResponse
+            (API.getFederationStatus u [d, d'])
+            $ \resp -> do
+              resp.status `shouldMatchInt` 200
+              resp.json %. "status" `shouldMatch` "fully-connected"
+    assertConnected uidA domainB domainC
+    assertConnected uidB domainA domainC
+    assertConnected uidC domainA domainB
+
+testDynamicBackendsNotFederating :: HasCallStack => App ()
+testDynamicBackendsNotFederating = do
+  let overrides =
+        def
+          { dbBrig =
+              setField "optSettings.setFederationStrategy" "allowDynamic"
+                >=> removeField "optSettings.setFederationDomainConfigs"
```

Comment:
```suggestion
              setField "optSettings.setFederationStrategy" "allowNone"
```

(same thing, but this one's shorter, and doesn't require cleaning the cassandra table.)